### DISCUSSION TITLE
Help on the procedure to upload the firmware to the STM32

### DISCUSSION THREAD

**Discussion by f616** [AxxAxx/AxxSolder#28] [Q&A] [answered]
I've been totally unable to connect the STLINK-V3MINIE and upload the firmware.
I've connected 3,3V and GND to an external power supply, have connected STLINK-V3MINIE RST, SWCLK, SWDIO to the board (common GNS with the powersupply), Nothing else connected (no LCD, no handle no any other external power).

 STLINK-V3MINIE driver have been installed on the system (windows 10)

![image](https://github.com/AxxAxx/AxxSolder/assets/3826426/5a419cf2-f593-46a2-872f-ebf67a72ae86)

When trying to connect, I'm always getting a `Error: No STM32 target found!`

![image](https://github.com/AxxAxx/AxxSolder/assets/3826426/eb708617-7120-490e-82c7-928a0f2b44fa)

Is there any clue I can get from you?
Can this be hardware related? Can you point out some clues where/what/how the componentes to measure?

**Comment by f616** [comment, marked as answer]
Issue solved.
I was missing the 3.3v connection back to STLINK-V3MINIE. It seems that the STLINK-V3MINIE was not (fully) powered through USB and also need to be powered through the external 3.3V power supply

**Comment by AxxAxx** [comment]
Yes indeed, the 3.3V (pin 10) T_VCC on the STLINK-V3MINIE is an INPUT and can not source any current. 

I have installed a 5V->3.3V LDO on my STLINK-V3MINIE:
![PXL_20231211_180450515](https://github.com/AxxAxx/AxxSolder/assets/12958429/a1a407eb-c9aa-4b46-bb05-cfe213a884cc)
